symphony-bdk-core/src/main/java/com/symphony/bdk/core/util/BdkExponentialFunction.java

```diff
@@ -0,0 +1,36 @@
+package com.symphony.bdk.core.util;
+
+import com.symphony.bdk.core.config.model.BdkRetryConfig;
+import io.github.resilience4j.core.IntervalFunction;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Custom BdkExponentialFunction class to be used in Bdk Retry mechanism.
+ */
+@Slf4j
+public class BdkExponentialFunction {
+
+    /**
+     * Produce an interval function from given {@link BdkRetryConfig}.
+     *
+     * @param retryConfig given retry configuration.
+     *
+     * @return an {@link IntervalFunction} to be used in Retry mechanism.
+     */
+    public static IntervalFunction ofExponentialBackoff(BdkRetryConfig retryConfig) {
```

Comment:
Remember that we created a dedicated story (cf. APP-2821) to properly manage retries.